atoma-proxy-service/src/components/grafana.rs

```diff
@@ -0,0 +1,207 @@
+        let to = dashboard.dashboard.time.to;
+        dashboard
+            .dashboard
+            .panels
+            .into_iter()
+            .map(|panel| {
+                (
+                    panel.title,
+                    Query {
+                        queries: panel.targets,
+                        from: from.clone(),
+                        to: to.clone(),
+                    },
+                )
+            })
+            .collect()
+    }
+}
+
+impl Grafana {
+    /// Create a new Grafana client
+    ///
+    /// # Arguments
+    ///
+    /// * `url` - The URL of the Grafana instance
+    /// * `api_token` - The API token to use for authentication
+    /// * `dashboard_tag` - The tag to use to filter dashboards
+    #[must_use]
+    pub const fn new(url: String, api_token: String, dashboard_tag: String) -> Self {
+        Self {
+            url,
+            api_token,
+            dashboard_tag,
+        }
+    }
+
+    /// Get the UIDs of all dashboards with the specified tag
+    ///
+    /// # Returns
+    ///
+    /// A vector of dashboard UIDs
+    #[instrument(level = "info", skip_all)]
+    pub async fn get_dashboard_uids(&self) -> Result<Vec<String>, GrafanaError> {
+        let client = Client::new();
+        let request_url = format!("{}/api/search?tag={}", self.url, self.dashboard_tag);
+        let response = client
+            .get(&request_url)
+            .header("Accept", "application/json")
+            .header("Content-Type", "application/json")
+            .header("Authorization", format!("Bearer {}", self.api_token))
+            .send()
+            .await?;
+
+        if response.status().is_success() {
+            let dashboards = response.json::<Vec<DashboardList>>().await?;
+            Ok(dashboards
+                .into_iter()
+                .map(|dashboard| dashboard.uid)
+                .collect())
+        } else {
+            Err(GrafanaError::FailedRequest(
+                response.error_for_status().unwrap_err(),
+            ))
+        }
+    }
+
+    /// Get a dashboard by its UID
+    ///
+    /// # Arguments
+    ///
+    /// * `dashboard_uid` - The UID of the dashboard to get
+    ///
+    /// # Returns
+    ///
+    /// The dashboard with the specified UID
+    #[instrument(level = "info", skip(self))]
+    pub async fn get_dashboard(&self, dashboard_uid: String) -> Result<Dashboard, GrafanaError> {
+        let client = Client::new();
+        let request_url = format!("{}/api/dashboards/uid/{}", self.url, dashboard_uid);
+        let response = client
+            .get(&request_url)
+            .header("Accept", "application/json")
+            .header("Authorization", format!("Bearer {}", self.api_token))
+            .send()
+            .await?;
+
+        if response.status().is_success() {
+            let dashboard = response.json::<Dashboard>().await?;
+            Ok(dashboard)
+        } else {
+            Err(GrafanaError::FailedRequest(
```

Comment:
We could perhaps make these errors a bit more granular as we do in other instances.

e.g.

```rust
pub enum GrafanaError {
    #[error("Request failed: {0}")]
    FailedRequest(#[from] reqwest::Error),
    #[error("Authentication failed")]
    AuthenticationError,
    #[error("Dashboard not found: {0}")]
    DashboardNotFound(String),
    #[error("Invalid response format: {0}")]
    InvalidResponse(#[from] serde_json::Error),
}
```

Reply:
I can do the status codes, but serde_json::Error is handled internally in the reqwest crate, I'm not going to convert it back